Rework the how-to about integration aiida-core components into WorkGraph

agoscinski opened the pull request:
Suggestions have been added as discussed.
- namespace related topics have been removed
- WorkChain, calcfunction, workfunction have been added
- Motivate more the beginning
- convert to jupytext python script (it takes  <6s to run the notebook )

agoscinski committed:
Rework the how-to about integration aiida-core components into WorkGraph

superstar54 reviewed (changes requested):
Hi @agoscinski, thanks for the work — most of it looks good to me! I have a few change requests:

* In the previous documentation, there is a section titled **“Use process builder.”** Please keep it, as the process builder is also widely used in AiiDA and remains important for users.
* Some sentences have grammatical issues. Please go through the text and revise — we aim to make the documentation as polished as possible for the stable release.
* Some parts have formatting issues in the preview, see the comments.

agoscinski committed:
Revision

- fix note; fix link at the end
- Add process builder back
- fix language here and there
- remove jupytext preamble
- print final result
- remove computer localhost creation

superstar54 reviewed (approved):
LGTM!

agoscinski committed:
Merge branch 'main' into rework-use-aiida-components
